### ISSUE TITLE
Can't build Node.js gRPC Library

### ISSUE THREAD

**Issue by jonywtf** [grpc/grpc#941]
I used latest io.js v1.4.3, but with node v0.12.0 can't to build too

cd ~/grpc/src/node
node-gyp build

```
gyp info it worked if it ends with ok
gyp info using node-gyp@1.0.2
gyp info using node@1.4.3 | linux | x64
child_process: customFds option is deprecated, use stdio instead.
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory `/root/grpc/src/node/build'
  CXX(target) Release/obj.target/grpc/ext/byte_buffer.o
In file included from ../ext/byte_buffer.cc:38:0:
../node_modules/nan/nan.h:481:19: error: ‘NanNew’ declared as an ‘inline’ variable
       v8::String::ExternalAsciiStringResource *resource) {
                   ^
../node_modules/nan/nan.h:481:19: warning: ‘always_inline’ attribute ignored [-Wattributes]
../node_modules/nan/nan.h:481:19: error: ‘v8::Local<v8::String> NanNew’ redeclared as different kind of symbol
../node_modules/nan/nan.h:475:36: error: previous declaration of ‘v8::Local<v8::String> NanNew(v8::String::ExternalStringResource*)’
   NAN_INLINE v8::Local<v8::String> NanNew(
                                    ^
../node_modules/nan/nan.h:481:7: error: ‘ExternalAsciiStringResource’ is not a member of ‘v8::String’
       v8::String::ExternalAsciiStringResource *resource) {
       ^
../node_modules/nan/nan.h:481:48: error: ‘resource’ was not declared in this scope
       v8::String::ExternalAsciiStringResource *resource) {
                                                ^
../node_modules/nan/nan.h: In function ‘bool _NanGetExternalParts(v8::Handle<v8::Value>, const char**, size_t*)’:
../node_modules/nan/nan.h:2103:12: error: ‘class v8::String’ has no member named ‘IsExternalAscii’
   if (str->IsExternalAscii()) {
            ^
../node_modules/nan/nan.h:2104:11: error: ‘ExternalAsciiStringResource’ in ‘class v8::String’ does not name a type
     const v8::String::ExternalAsciiStringResource* ext;
           ^
../node_modules/nan/nan.h:2105:5: error: ‘ext’ was not declared in this scope
     ext = str->GetExternalAsciiStringResource();
     ^
../node_modules/nan/nan.h:2105:16: error: ‘class v8::String’ has no member named ‘GetExternalAsciiStringResource’
     ext = str->GetExternalAsciiStringResource();
                ^
../ext/byte_buffer.cc: At global scope:
../ext/byte_buffer.cc:47:15: error: namespace ‘node::Buffer’ not allowed in using-declaration
 using ::node::Buffer;
               ^
../ext/byte_buffer.cc: In function ‘grpc_byte_buffer* grpc::node::BufferToByteBuffer(v8::Handle<v8::Value>)’:
../ext/byte_buffer.cc:57:16: error: ‘Buffer’ has not been declared
   int length = Buffer::Length(buffer);
                ^
../ext/byte_buffer.cc:58:16: error: ‘Buffer’ has not been declared
   char *data = Buffer::Data(buffer);
                ^
In file included from ../ext/byte_buffer.cc:38:0:
../ext/byte_buffer.cc: In function ‘v8::Handle<v8::Value> grpc::node::ByteBufferToBuffer(grpc_byte_buffer*)’:
../node_modules/nan/nan.h:504:33: error: ‘args’ was not declared in this scope
 # define NanReturnNull() return args.GetReturnValue().SetNull()
                                 ^
../ext/byte_buffer.cc:69:5: note: in expansion of macro ‘NanReturnNull’
     NanReturnNull();
     ^
../ext/byte_buffer.cc: In function ‘v8::Handle<v8::Value> grpc::node::MakeFastBuffer(v8::Handle<v8::Value>)’:
../ext/byte_buffer.cc:85:30: error: ‘GetCurrent’ is not a member of ‘v8::Context’
   Handle<Object> globalObj = Context::GetCurrent()->Global();
                              ^
../ext/byte_buffer.cc:89:48: error: ‘Buffer’ has not been declared
                                 NanNew<Number>(Buffer::Length(slowBuffer)),
                                                ^
make: *** [Release/obj.target/grpc/ext/byte_buffer.o] Error 1
make: Leaving directory `/root/grpc/src/node/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:169:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1044:12)
gyp ERR! System Linux 3.13.0-43-generic
gyp ERR! command "node" "/usr/local/bin/node-gyp" "build"
gyp ERR! cwd /root/grpc/src/node
gyp ERR! node -v v1.4.3
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok
```

Please, help me! Which version should I use?

**Comment by murgatroid99** [comment]
I just published grpc 0.5.2 to npm. It should build on all versions of io.js and Node 0.10.13+. It currently crashes when it finishes on Node 0.11+, which I am investigating in issue #958.

**Comment by jonywtf** [comment]
@murgatroid99, Thank you very much!!!
